### DISCUSSION TITLE
Client.get_user_tweets() fails to retrieve protected tweets, even when authenticated user account has been allowed to view these tweets

### DISCUSSION THREAD

**Discussion by dehueck** [tweepy/tweepy#2071] [Questions] [answered]
Running the code below where `user_id` is that of a Twitter account with protected tweets that has allowed user whose authentication gives us `access_token` and `access_token_secret` below returns a "Not authorized for resource error":

`Response(data=None, includes={}, errors=[{'resource_id': 'XXXXXXXXXXXXXXXXX, 'parameter': 'id', 'resource_type': 'user', 'section': 'data', 'title': 'Authorization Error', 'value': 'XXXXXXXXXXXXXXXXX', 'detail': 'Sorry, you are not authorized to see the user with id: [XXXXXXXXXXXXXXXX].', 'type': 'https://api.twitter.com/2/problems/not-authorized-for-resource'}], meta={})`

Raising the issue because I was not experiencing the problem about ~2 months ago & wondering if recent changes to API have affected some tweepy functionality. Any thoughts? Thanks--

`

```
def get_tweets(user_id):
    
    auth = tweepy.OAuth1UserHandler(api_key, api_key_secret, access_token, access_token_secret)
    api = tweepy.API(auth, wait_on_rate_limit=True)

    client = tweepy.Client(
        bearer_token=bearer_token, 
        consumer_key=api_key, 
        consumer_secret=api_key_secret, 
        access_token=access_token, 
        access_token_secret=access_token_secret)


    start_time = "2022-12-29T10:00:01Z". # this is an arbitrary start date

    for tweet in tweepy.Paginator(client.get_users_tweets, user_id, start_time=start_time):
        print(tweet)
````

**Comment by Harmon758** [comment, marked as answer]
By default, [`Client.get_users_tweets`](https://tweepy.readthedocs.io/en/v4.12.1/client.html#tweepy.Client.get_users_tweets) does not use OAuth 1.0a User Context to authenticate, so if you're passing an app-only bearer token, it's attempting to use that.

You need to pass `False` for `user_auth` or use an OAuth 2.0 Authorization Code Flow with PKCE access token.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Reply by dehueck** [reply]
wow ok thank you I think that's it. appreciate the help sorry new to GitHub so I probably mistakenly raised as an issue instead of something else...

**Reply by dehueck** [reply]
if i have a suggestion about revising one small section of the docs (for clarity's sake), where should I post that?

**Reply by Harmon758** [reply]
You can just make a PR, but I would suggest running it by me first.
Here is fine if it's relevant, or we can talk about it in [the Tweepy Discord server](https://discord.gg/bJvqnhg).

**Reply by dehueck** [reply]
Ok I've done a little bit of tinkering in an "app-only" kind of setting (not using any user info) & just wanted to be sure I haven't misunderstood varieties of auth-flows as the following is still returning an error for me.

```
import tweepy
bearer_token = XXXXXXXXXXXXXXX

def get_tweets(user_id):
    
    client=tweepy.Client(bearer_token)

    start_time = "2022-12-29T10:00:01Z"

    for tweet in client.get_users_tweets(user_id, start_time=start_time, user_auth=False):
        print(tweet)
```

Based on this section -- https://docs.tweepy.org/en/stable/authentication.html#id2 (OAuth 2.0 Bearer Token (App-Only), I think I've implemented this correctly but am still getting the following error:

`[{'resource_id': 'XXXXX', 'parameter': 'id', 'resource_type': 'user', 'section': 'data', 'title': 'Authorization Error', 'value': 'XXXXXX', 'detail': 'Sorry, you are not authorized to see the user with id: [XXXXXX].', 'type': 'https://api.twitter.com/2/problems/not-authorized-for-resource'}]`

I am using bearer token from developer account, trying to access tweets that the account linked to my developer account should be able to see. Thanks--

**Reply by dehueck** [reply]
Continuing to read more -- is this because I am authenticating as an app and not as a user? Was going to try OAuth 2.0 Authorization Code Flow with PKCE shortly

**Reply by dehueck** [reply]
I think I may be making the mistake of assuming that doing app-only auth w/bearer token will give me privileges associated with account my developer account is registered under.

**Reply by Harmon758** [reply]
Please read the answer.
